app/collection.go

```diff
@@ -17,29 +16,20 @@ func (a *App) registerCollectionAndTopic(pluginID, collectionType, topicType str
 	defer a.ch.collectionAndTopicTypesMut.Unlock()
 
 	// check if collectionType was already registered by other plugin
-	for existingPluginID, existingCollectionTypes := range a.ch.collectionTypes {
-		if existingPluginID != pluginID && utils.StringInSlice(collectionType, existingCollectionTypes) {
-			return model.NewAppError("registerCollectionAndTopic", "app.collection.add_collection.exists.app_error", nil, "", http.StatusBadRequest)
-		}
+	existingPluginID, ok := a.ch.collectionTypes[collectionType]
+	if existingPluginID != pluginID && ok {
```

Comment:
For clarity, perhaps reversing the check?
```suggestion
	existingPluginID, ok := a.ch.collectionTypes[collectionType]
	if ok && existingPluginID != pluginID {
```